### ISSUE TITLE
数据生成函数为什么最后要返回一个np.ones(batch_size)？

### ISSUE THREAD

**Issue by Yogaht** [ypwhs/captcha_break#17]
yield [X, y, np.ones(batch_size)*int(conv_shape[1]-2), np.ones(batch_size)*n_len],  **np.ones(batch_size)**

为什么这里要返回一个np.ones(batch_size）是为了和fit_generrate要求的输入匹配吗，不返回label也不会出现问题吗？

**Comment by Yogaht** [comment]
请教一下，谢谢

**Comment by ypwhs** [comment]
这是 Keras 的限制，Keras 要求训练的时候必须有一个输入和一个标签，通过优化输出和标签的差别来进行训练。但是我们这个问题里面要求输入 label、label_length、input_length，所以只能在 base_model 之外再搭建一个专门用于训练的模型，这个模型只输出 loss：`model.compile(loss={'ctc': lambda y_true, y_pred: y_pred}, optimizer='adadelta')` 因此这里的 y_true 是无意义的，只需要生成尺寸一致的矩阵就能通过 Keras 的检验。

**Comment by RenXiandong888** [comment]
int(conv_shape[1]-2)
能问一下，这个地方为什么要减去2么？
